q2_diversity/_alpha.py

```diff
@@ -5,3 +5,110 @@
 #
 # The full license is in the file LICENSE, distributed with this software.
 # ----------------------------------------------------------------------------
+
+import os
+
+import biom
+import scipy
+import skbio
+import qiime
+import pandas as pd
+import seaborn as sns
+import skbio.diversity
+from statsmodels.sandbox.stats.multicomp import multipletests
+
+
+# We should consider moving these functions to scikit-bio. They're part of
+# the private API here for now.
+def phylogenetic_metrics():
+    return {'faith_pd'}
+
+
+def non_phylogenetic_metrics():
+    return {'ace', 'chao1', 'chao1_ci', 'berger_parker_d', 'brillouin_d',
+            'dominance', 'doubles', 'enspie', 'esty_ci', 'fisher_alpha',
+            'goods_coverage', 'heip_e', 'kempton_taylor_q', 'margalef',
+            'mcintosh_d', 'mcintosh_e', 'menhinick', 'michaelis_menten_fit',
+            'observed_otus', 'osd', 'pielou_e', 'robbins', 'shannon',
+            'simpson', 'simpson_e', 'singles', 'strong', 'gini_index',
+            'lladser_pe', 'lladser_ci'}
+
+
+def alpha_phylogenetic(metric: str, feature_table: biom.Table,
+                       phylogeny: skbio.TreeNode=None) -> pd.Series:
+    if metric not in phylogenetic_metrics():
+        raise ValueError("Unknown phylogenetic metric: %s" % metric)
+
+    counts = feature_table.matrix_data.toarray().astype(int).T
+    sample_ids = feature_table.ids(axis='sample')
+    feature_ids = feature_table.ids(axis='observation')
+
+    return skbio.diversity.alpha_diversity(
+              metric=metric,
+              counts=counts,
+              ids=sample_ids,
+              otu_ids=feature_ids,
+              tree=phylogeny)
+
+
+def alpha(metric: str, feature_table: biom.Table) -> pd.Series:
+    if metric not in non_phylogenetic_metrics():
+        raise ValueError("Unknown metric: %s" % metric)
+
+    counts = feature_table.matrix_data.toarray().astype(int).T
+    sample_ids = feature_table.ids(axis='sample')
+
+    return skbio.diversity.alpha_diversity(
+              metric=metric,
+              counts=counts,
+              ids=sample_ids)
+
+
+def alpha_comparisons(output_dir: str, alpha_diversities: pd.Series,
```

Comment:
How about `alpha_compare`? Shorter name and closer to qiime1 script name.

How do you feel about renaming `alpha_diversities` to `alpha_diversity`? That way the name matches the semantic type name and the named output of `alpha` and `alpha_phylogenetic`. I was confused by this plural name, I initally thought it was composed of more than one `alpha_diversity` result e.g. that was concatenated.